introspection/providers_tree.go

```diff
@@ -0,0 +1,58 @@
+package introspection
+
+// ProvidersTree is a struct which provider modules use to register their interest in providing
+// data to be recorded in a time slice of the running system.
+type ProvidersTree struct {
+	Runtime *RuntimeProviders
```

Comment:
I think we can vastly simplify this model by passing in an object to express the type of query we want. If you take a step back and look at the wider pattern here, both `Get`/`List` are really resolver functions that return the same type; what varies is the type of the query, and we don't need to splinter query types into dedicated functions.

```go
type ProvidersMap struct {
    Runtime    func()
    Connection func(ConnectionQueryInput) ([]*Conn, error)
    Stream     func(StreamQueryInput) ([]*Stream, error)
}
```

Reply:
This is a really good idea & as I've gradually begun to understand -> also in the spirit of GraphQL resolvers. 

I've made this change. Also, I've removed the Stream resolver for now as for **0.1**, the streams will be resolved by the connection resolver & the WS client never explicitly asks us to resolve Streams. We can always add it back later when we need it. 

Let me know what you think about the refactored code.